ENIGMAsystem/SHELL/Platforms/General/POSIX/POSIXdirectory.cpp

```diff
@@ -34,8 +51,13 @@ std::string filename_absolute(std::string fname) {
   if (file_exists(fname) || directory_exists(fname)) {
     char rpath[PATH_MAX];
     char *ptr = realpath(fname.c_str(), rpath);
-    if (ptr != NULL) { return rpath; }
-  } return "";
+    if (ptr != NULL) return add_slash(ptr);
```

Comment:
This function used to support canonicalizing filenames. Now it converts filenames into directories.

Reply:
@JoshDreamland fix it